lib/OpenQA/Utils.pm

```diff
@@ -425,28 +393,18 @@ sub find_labels {
     return \@labels;
 }
 
-sub find_flags ($text) {
-    my @flags;
-    push @flags, $+{match} while $text =~ /${\FLAG_REGEX}/g;
-    return \@flags;
-}
-
-sub find_bugref {
-    my ($text) = @_;
-    $text //= '';
+sub find_bugref ($text = '') {
     $text =~ BUGREF_REGEX;
     return $+{match};
 }
 
-sub find_bugrefs {
-    my $text = shift // '';
+sub find_bugrefs ($text = '') {
```

Comment:
That would allow calling `find_bugrefs` without an argument. That doesn't make sense to me.

Reply:
even before the subroutine could be called without arguments

Reply:
Yes, because we didn't use signatures, so there was no check. But now with signatures we can check that, so calls without arguments will get a clear error message. That's one of the advantages of signatures.

Reply:
so previously when code would have called `find_bugrefs $undefined_object` that would just return falso from the fanction as there is no match. How can we keep that functionality? Do I need to do

```perl
sub find_bugrefs ($text = undef) {
    $text //= '';
```

Reply:
@perlpunk should I do what I suggested in the above code example?